activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -107,7 +107,8 @@ def take!
     end
 
     # Find the first record (or first N records if a parameter is supplied).
-    # If no order is defined it will order by primary key.
+    # If no order is defined it will order by primary key if the collection is not loaded.
+    # If the collection is loaded it will get the first record from that collection.
```

Comment:
@sgrif take a look at #24624 . It seems like some methods behavior depends on the load status of the relation. And this methods can return totaly different results dependent on this state. 

I think it is difficult to describe when a relation gets the state loaded for some one not familiar with AR. Maybe methods which lead to a state "loaded" should be marked so that some one less familiar with AR can recognize them and docs of a method with different behavior can reference methods with a "loaded" site effect.